src/Nethermind/Nethermind.Specs/ChainSpecStyle/ChainSpecBasedSpecProvider.cs

```diff
@@ -38,58 +40,88 @@ public ChainSpecBasedSpecProvider(ChainSpec chainSpec)
 
         private void BuildTransitions()
         {
-            SortedSet<long> transitionBlocks = new();
-            transitionBlocks.Add(0L);
+            SortedSet<long> transitionBlockNumbers = new();
+            SortedSet<ulong> transitionTimestamps = new();
+            transitionBlockNumbers.Add(0L);
 
             if (_chainSpec.Ethash?.BlockRewards != null)
             {
                 foreach ((long blockNumber, _) in _chainSpec.Ethash.BlockRewards)
                 {
-                    transitionBlocks.Add(blockNumber);
+                    transitionBlockNumbers.Add(blockNumber);
                 }
             }
 
-            var forks = _chainSpec.GetType()
+            var blockNumberforks = _chainSpec.GetType()
                 .GetProperties(BindingFlags.Public | BindingFlags.Instance)
-                .Where(p => p.Name.Contains("BlockNumber") && p.Name != "TerminalPoWBlockNumber");
+                .Where(p => p.Name.EndsWith("BlockNumber") && p.Name != "TerminalPoWBlockNumber");
+
+            var timestampForks = _chainSpec.GetType()
```

Comment:
p.Name.EndsWith("Timestamp") hmmm just wondering, because we could have field called with Timestamp right now. In theory we could leave the naming convention "Transition" in chainspec

Reply:
it is possible but would be confusing in my opinion.